src/Input/Select/Select.numberChecking.test.tsx

```diff
@@ -0,0 +1,123 @@
+        <Select.Option value="one">{1}</Select.Option>
+        <Select.Option value="two">{2}</Select.Option>
+        <Select.Option value="three">{3}</Select.Option>
+      </Select>
+    );
+
+    const { queryByRole, queryAllByTestId, asFragment } = render(<Component />);
+
+    const selectInput = queryByRole('combobox');
+    userEvent.type(selectInput, '1');
+    const filteredOptions = queryAllByTestId('option');
+    expect(filteredOptions.length).toEqual(1);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('number defaultValue', async () => {
+    const testNumberDefaultValue = ([defaultValue, expectedValue]: any) => {
+      const Component = () => (
+        <Select label="number defaultValue" defaultValue={defaultValue}>
+          <Select.Option value="one">{1}</Select.Option>
+          <Select.Option value="two">{2}</Select.Option>
+          <Select.Option value="three">{3}</Select.Option>
+        </Select>
+      );
+
+      const { queryByRole, unmount } = render(<Component />);
+      const selectInput = queryByRole('combobox') as HTMLInputElement;
+      expect(selectInput.value).toEqual(expectedValue);
+      unmount();
+    };
+
+    [
+      [1, '1'],
+      [{ key: 'value' }, '[object Object]'],
+      [[1, 2, 3], '1,2,3'],
+    ].forEach(([defaultValue, expectedValue]) => {
+      testNumberDefaultValue([defaultValue, expectedValue]);
+    });
+  });
+
+  test('number value', async () => {
+    const testNumberValue = ([value, expectedValue]: any) => {
+      const Component = () => (
+        <Select label="number defaultValue" value={value}>
+          <Select.Option value="one">{1}</Select.Option>
+          <Select.Option value="two">{2}</Select.Option>
+          <Select.Option value="three">{3}</Select.Option>
+        </Select>
+      );
+
+      const { queryByRole, unmount } = render(<Component />);
+      const selectInput = queryByRole('combobox') as HTMLInputElement;
+      expect(selectInput.value).toEqual(expectedValue);
+      unmount();
+    };
+
+    [
+      [1, '1'],
+      [{ key: 'value' }, '[object Object]'],
+      [[1, 2, 3], '1,2,3'],
+    ].forEach(([value, expectedValue]) => {
+      testNumberValue([value, expectedValue]);
+    });
+  });
+
+  test('rerender with number value', async () => {
+    const Component = ({ value }: { value: string }) => (
+      <Select label="rerender" value={value}>
+        <Select.Option value="rerender">Rerender</Select.Option>
+      </Select>
+    );
+
+    const { queryByRole, rerender } = render(<Component value="Rerender" />);
+    const selectInput = queryByRole('combobox');
+    expect(selectInput.getAttribute('value')).toEqual('Rerender');
+
+    rerender(<Component value={1} />);
+    expect(selectInput.getAttribute('value')).toEqual('1');
+
+    rerender(<Component value={[1, 2, 3]} />);
+    expect(selectInput.getAttribute('value')).toEqual('1,2,3');
+
+    rerender(<Component value={{ key: 'value' }} />);
```

Comment:
same comment for this one.
https://github.com/glints-dev/glints-aries/pull/996/files#r1424903960